pilot/pkg/security/authz/matcher/header.go

```diff
@@ -36,22 +36,34 @@ func HeaderMatcher(k, v string) *routepb.HeaderMatcher {
 	} else if strings.HasPrefix(v, "*") {
 		return &routepb.HeaderMatcher{
 			Name: k,
-			HeaderMatchSpecifier: &routepb.HeaderMatcher_SuffixMatch{
-				SuffixMatch: v[1:],
+			HeaderMatchSpecifier: &routepb.HeaderMatcher_StringMatch{
```

Comment:
Does the PR change the current header matching behaviors? Or the PR replaces the deprecated APIs with newer equivalent APIs with no behavior changes?

Reply:
There aren't any behavior changes. They just changed to the API. This has been in Envoy for a few years, but we've never changed this.

Reply:
Thanks for the PR! LGTM.